### DISCUSSION TITLE
tutorial

### DISCUSSION THREAD

**Discussion by TOcvfan** [vercel/next.js#50650] [Help] [answered]
### Summary

I have been following the official next.js tutorial so that I can learn it
I checked the code and it's the same as in the tutorial, but there's a big difference from mine and the one in the tutorial
![image](https://github.com/vercel/next.js/assets/11290288/559277f4-e560-4e08-b24a-da4bb35b2d19)
![image](https://github.com/vercel/next.js/assets/11290288/8d538fcc-a1c6-4023-bfe6-53a03183dfc6)

there's also a difference in the front page
![image](https://github.com/vercel/next.js/assets/11290288/0e68bb3f-a0d1-4e6b-af16-537fd8e1b927)
![image](https://github.com/vercel/next.js/assets/11290288/5d13bc5f-137c-4f76-9266-2fbbabd8b04c)

I used console.log in pages/index.js file
![image](https://github.com/vercel/next.js/assets/11290288/d399568d-12d1-4261-a895-62d967c4a0b7)

**Comment by devjiwonchoi** [comment]
Could you elaborate on exactly which part do you need for help?

Please provide the URL of the docs and the link to the source code you are referring.

**Reply by TOcvfan** [reply]
It's these parts of the tutorial that's giving me problems 
https://nextjs.org/learn/basics/data-fetching/implement-getstaticprops
https://nextjs.org/learn/basics/dynamic-routes/implement-getstaticprops

I'm not sure which source code you mean, if you mean for the tutorial it's this one:
https://github.com/vercel/next-learn/tree/master/basics/dynamic-routes-step-1

**Reply by devjiwonchoi** [reply]
I'm assuming that it's an installation problem or an import path.

Could you share your forked repo in order to give you a more precise and faster answer?

**Reply by TOcvfan** [reply]
https://github.com/TOcvfan/nextapp

**Reply by devjiwonchoi** [reply]
I'm assuming that you've forked [`vercel/next-learn`](https://github.com/vercel/next-learn), and your question is that your local deployment web view is not identical with the [example app](https://next-learn-starter.vercel.app/).

> I checked the code and it's the same as in the tutorial, but there's a big difference from mine and the one in the tutorial

Is that correct?

If so, I want to see the `code` that you have mentioned that differs from the tutorial.

**Reply by TOcvfan** [reply]
No I used this command as the tutorial said
`npx create-next-app@latest nextjs-blog --use-npm --example "https://github.com/vercel/next-learn/tree/master/basics/learn-starter"`

And I followed the instructions in the tutorial, but I didn't get the same result as shown in the screenshots

**Reply by devjiwonchoi** [reply]
If you haven't pushed it to your [repository](https://github.com/TOcvfan/nextapp), please do so and let me check the codes you are talking about.

![image](https://github.com/vercel/next.js/assets/120007119/213d2c2f-0c92-4526-960d-fcec799880d6)
These codes.

**Reply by TOcvfan** [reply]
> https://github.com/TOcvfan/nextapp

It's this one..

**Reply by devjiwonchoi** [reply]
I'm sure it's not. Please double-check.
<img width="926" alt="Screenshot 2023-06-02 at 10 05 26 PM" src="https://github.com/vercel/next.js/assets/120007119/e10480a4-2cea-48e5-b99c-3eab2b51b2c4">

**Reply by TOcvfan** [reply]
sorry I thought that it would add all the changes I made when I remote added it (which I did today not 2 days ago), but it didn't.
It's there now

**Comment by devjiwonchoi** [comment, marked as answer]
From your code from `/lib/posts.js`,
```js
const postsDirectory = path.join(process.cwd(), 'pages/posts');
```
the `posts` folder was inside the `pages` directory, and Next.js was understanding it as a `route`.

> When a file is added to the pages directory, it's automatically available as a route.

For more information, look [here](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts).

All you need to do is put the `posts` folder with `pre-rendering.md` and `ssg-ssr.md` in the root directory.

![Screenshot 2023-06-03 at 12 44 32 AM](https://github.com/vercel/next.js/assets/120007119/126cd7cb-f47f-44eb-aa66-8d8997f13415)

Also don't forget to change your import on your `posts.js`.

```js
const postsDirectory = path.join(process.cwd(), 'posts');
```

I'll open a PR to your repository for you to look at.

**Reply by TOcvfan** [reply]
thanks I didn't see that it was at top-lvl it was supposed to be at also i found out that the md files was incorrect, it had ```md at the top of the file and ``` at the bottom which did that the file could not be read right.
Thank you for the help

**Reply by devjiwonchoi** [reply]
I just found out that you've already fixed it. Cheers! 🍻
